core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -176,6 +190,25 @@ private class ClientEndpoint(
       } else if (!Utils.responseFromBackup(message)) {
         System.exit(-1)
       }
+
+    case DriverStatusResponse(found, state, _, _, _) =>
+      if (found) {
+        state.get match {
+          case DriverState.FINISHED | DriverState.FAILED |
+               DriverState.ERROR | DriverState.KILLED =>
+            logInfo(s"State of $submittedDriverID is ${state.get}, " +
+              s"exiting spark-submit JVM.")
+            System.exit(0)
+          case _ =>
+            Thread.sleep(REPORT_DRIVER_STATUS_INTERVAL)
```

Comment:
`Thread.sleep` could still has the same issue, imaging the network drop happens during sleeping. We should control the period sending logic out of `receive`. We could mimic `CoarseGrainedSchedulerBackend` to do the same work here:

https://github.com/apache/spark/blob/9faad07ce706890008a8a3ce675fa95b0bdf7c14/core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala#L137-L139

Reply:
@Ngone51 Thanks for reviewing. I tried simulating that case by increasing the `Thread.sleep `time and I see that `onDisconnected` is called when the master is stopped during sleep time and following logs are generated:
```
20/05/16 13:24:44 ERROR ClientEndpoint: Error connecting to master 127.0.0.1:7077.
20/05/16 13:24:44 ERROR ClientEndpoint: No master is available, exiting.
```

Reply:
@Ngone51 Thanks for reviewing. I have updated to use the task scheduler to do the same. Kindly review it again.